### ISSUE TITLE
question: general wlroots porting plan?

### ISSUE THREAD

**Issue by acrisci** [Immington-Industries/way-cooler#529]
Do you have a general plan for how to start doing the wlroots part of the new way-cooler?

On sway, we decided to do rootston first to work out the kinks in the library, provide an example on how to use the library, and to be sort of like integration tests as we developed new features. We always kept the sway use case in mind as we made rootston.

I see that wlroots-rs started to have a rootston project, but it was removed. Do you want to go that route with it, or go straight to making way-cooler?

I'm sure you've thought about this, so I'd like to hear what you think about it before I start thinking about how to structure the code.

**Comment by Timidger** [comment]
I started rootston and then realized it would be too much work to emulate the true rootston, especially now that it is used as a testbed for new features in wlroots.

I tried looking at the seat code for rootston and it was _very_ complicated and hard to follow. I wanted to make sure I had it 1-1, so I abandoned it because it would be a waste of effort since Way Cooler will not be multi-seat (for 1.0, at the very least).

If someone else wanted to start writing rootston in Rust then I'm not against it, it's just not something I'm interested in working on right now.

My personal plan is to work on Way Cooler since the API is actually pretty good for the most part (and there's really not much I can do to make it more ergonomic since I'm constrained by safety and the actual API) and fill in the incomplete things as I get to them. Onec wlroots 1.0 is released I'll focus on getting wlroots-rs done so I can release it in lock-step on crates.io.

In terms of code structuring right now I'm keeping all the awesome stuff in `awesome` and all the compositor stuff in `compositor`. There's overlap, but it's working really well. That means work can be done mostly in parallel without one drastically affecting the other (though obviously they need to work in concert with each other).
